test/powershell/Language/Scripting/HashtableToPSCustomObjectConversion.Tests.ps1

```diff
@@ -0,0 +1,248 @@
+    }
+       
+
+    Context 'Creating an object of an existing type from hashtable should succeed' {       
+       {
+                $source = @"
+                        public class SampleClass1
+                        {                                                       
+                            public SampleClass1(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass1()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $script:x = [SampleClass1]@{a=10;b=5}            
+       } | Should Not Throw
+       It '$x is not $null' { $script:x | Should Not Be $null }
+       It '$x.a' { $script:x.a | Should Be '10' }
+   }
+       
+
+    It 'Creating an object of an existing type from hashtable should throw error when setting non-existent properties' {
+      
+       try
+       { 
+          $source = @"
+                        public class SampleClass2
+                        {
+                            public SampleClass2(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass2()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass2]@{blah=10;b=5 }
+                 Throw "Exception expected, execution should not have reached here"
+       }
+       catch
+       {
+           $_.Exception.InnerException.ErrorRecord.FullyQualifiedErrorId | Should Be 'ObjectCreationError'
+       }       
+    }
+
+
+    It 'Creating an object of an existing type from hashtable should throw error when setting incompatible values for properties' {       
+       
+       try
+       { 
+          $source = @"
+                        public class SampleClass3
+                        {
+                            public SampleClass3(int x)
+                            {
+                            	a = x;
+                            }
+                            
+                            public SampleClass3()
+                            {
+                            }
+                                                          
+                            public int a;
+                            public int b;
+                        }
+"@
+                 add-type -typedefinition $source
+                 $x = [SampleClass3]@{a="foo";b=5}

```

Comment:
fixed

---
In reply to: [78255769](https://github.com/Powershell/Powershell/pull/2160#discussion_r78255769) [](ancestors = 78255769)